pkg/gui/filetree/inode.go

```diff
@@ -27,19 +25,17 @@ func sortChildren(node INode) {
 		return
 	}
 
-	children := node.GetChildren()
-	sortedChildren := make([]INode, len(children))
-	copy(sortedChildren, children)
+	sortedChildren := slices.Clone(node.GetChildren())
 
-	sort.Slice(sortedChildren, func(i, j int) bool {
-		if !sortedChildren[i].IsLeaf() && sortedChildren[j].IsLeaf() {
+	slices.SortFunc(sortedChildren, func(a, b INode) bool {
```

Comment:
one benefit of slices.SortFunc over the old sort.Slice is that it doesn't require a closure so you can extract out the callback into a named function (though I didn't feel the need to do that here)